torch/fx/experimental/symbolic_shapes.py

```diff
@@ -432,20 +432,31 @@ def _reduce_to_lowest_terms(expr: sympy.Expr) -> sympy.Expr:
     Useful when an expression is == or != to 0.
     """
     def integer_coefficient(x):
-        if isinstance(x, sympy.Integer):
+        if x.is_Integer:
```

Comment:
It's surprising to me that the attribute test is faster than isinstance; is this because there's some metaclass shenanigans in sympy or something?

Reply:
`isinstance` is slow regardless of metaclass complexity. If you know you have a sympy.Expr, attribute test is faster.